app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -218,52 +222,54 @@ class LuciBuildService {
       }
     }
 
-    return builderNames;
+    return targets;
   }
 
-  Future<List<String>> _builderNamesForRepo(
-    List<LuciBuilder> builders,
-    github.PullRequest pullRequest,
-  ) async {
-    final Map<String?, Build?> builds = await tryBuildsForPullRequest(pullRequest);
-    if (builds.values.any((Build? build) {
-      return build?.status == Status.scheduled || build?.status == Status.started;
-    })) {
+  /// List of targets that are not running.
+  ///
+  /// Throws [InternalServerError] if any target is scheduled or running on [pullRequest].
+  Future<List<Target>> _targetsNotScheduled(List<Target> targets, github.PullRequest pullRequest) async {
+    if (targets.isEmpty) {
+      throw InternalServerError('${pullRequest.base!.repo!.slug()} does not have any targets');
+    }
+
+    final Map<String?, Build?> tryBuilds = await tryBuildsForPullRequest(pullRequest);
+    if (tryBuilds.values.any((Build? build) => build?.status == Status.scheduled || build?.status == Status.started)) {
```

Comment:
This sounds a good place to retry only `queued` builds. Instead of return empty list, we return the unscheduled targets.

Reply:
Added!